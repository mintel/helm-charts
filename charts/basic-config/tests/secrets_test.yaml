suite: Test Secrets
templates:
  - secrets.yaml
release:
  namespace: test-namespace
tests:
  - it: Check renders a single external secret with no overrides
    set:
      global.name: test-app
      global.clusterEnv: qa
      extraSecrets:
        - name: "external-secret"
          path: test/external/secret
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: ExternalSecret
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: test-app-external-secret
      - equal:
          path: spec.dataFrom[0].key
          value: test-namespace/test-app/external-secret
      - hasDocuments:
          count: 1

  - it: Check secret name override
    set:
      global.name: test-app
      global.clusterEnv: qa
      extraSecrets:
        - nameOverride: "external-secret1"
          pathOverride: test/external/secret1
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - hasDocuments:
          count: 1
      - isKind:
          of: ExternalSecret
        documentIndex: 0
      - equal:
          path: metadata.name
          value: external-secret1
        documentIndex: 0

  - it: Check secret path override
    set:
      global.name: test-app
      global.clusterEnv: qa
      extraSecrets:
        - name: "external-secret1"
          pathOverride: test/external/secret1
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - hasDocuments:
          count: 1
      - isKind:
          of: ExternalSecret
        documentIndex: 0
      - equal:
          path: spec.dataFrom[0].key
          value: test/external/secret1
        documentIndex: 0

  - it: Check renders a single external secret
    set:
      global.name: test-app
      global.clusterEnv: qa
      extraSecrets:
        - name: "external-secret"
          pathOverride: test/external/secret
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: ExternalSecret
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: test-app-external-secret
      - equal:
          path: spec.dataFrom[0].key
          value: test/external/secret
      - equal:
          path: spec.secretStoreRef.kind
          value: SecretStore
      - hasDocuments:
          count: 1
      - equal:
          path: spec.secretStoreRef.name
          value: aws-secrets-manager-local
  - it: Check renders multiple external secrets
    set:
      global.name: test-app
      global.clusterEnv: qa
      extraSecrets:
        - name: "external-secret1"
          pathOverride: test/external/secret1
        - name: "external-secret2"
          pathOverride: test/external/secret2
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - hasDocuments:
          count: 2
      - isKind:
          of: ExternalSecret
        documentIndex: 0
      - isKind:
          of: ExternalSecret
        documentIndex: 1
      - equal:
          path: metadata.namespace
          value: test-namespace
        documentIndex: 0
      - equal:
         path: metadata.namespace
         value: test-namespace
        documentIndex: 1
      - equal:
          path: metadata.name
          value: test-app-external-secret1
        documentIndex: 0
      - equal:
          path: metadata.name
          value: test-app-external-secret2
        documentIndex: 1
      - equal:
          path: spec.dataFrom[0].key
          value: test/external/secret1
        documentIndex: 0
      - equal:
          path: spec.dataFrom[0].key
          value: test/external/secret2
        documentIndex: 1
      - equal:
          path: spec.secretStoreRef.kind
          value: SecretStore
        documentIndex: 0
      - equal:
          path: spec.secretStoreRef.kind
          value: SecretStore
        documentIndex: 1
      - equal:
          path: spec.secretStoreRef.name
          value: aws-secrets-manager-local
        documentIndex: 0
      - equal:
          path: spec.secretStoreRef.name
          value: aws-secrets-manager-local
        documentIndex: 1
  - it: Check secret name override
    set:
      global.name: test-app
      global.clusterEnv: qa
      extraSecrets:
        - nameOverride: "external-secret1"
          pathOverride: test/external/secret1
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - hasDocuments:
          count: 1
      - isKind:
          of: ExternalSecret
        documentIndex: 0
      - equal:
          path: metadata.name
          value: external-secret1
        documentIndex: 0
  - it: Check secret path override
    set:
      global.name: test-app
      global.clusterEnv: qa
      extraSecrets:
        - name: "external-secret1"
          pathOverride: test/external/secret1
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - hasDocuments:
          count: 1
      - isKind:
          of: ExternalSecret
        documentIndex: 0
      - equal:
          path: spec.dataFrom[0].key
          value: test/external/secret1
        documentIndex: 0
