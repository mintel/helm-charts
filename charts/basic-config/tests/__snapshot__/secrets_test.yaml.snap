Check renders a single external secret:
  1: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        helm.sh/chart: basic-config-0.2.3
      labels:
        app.kubernetes.io/managed-by: Helm
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app-external-secret
      name: test-app-external-secret
      namespace: test-namespace
    spec:
      dataFrom:
        - extract:
            key: test/external/secret
      refreshInterval: 5m0s
      secretStoreRef:
        kind: SecretStore
        name: aws-secrets-manager-local
      target:
        creationPolicy: Owner
Check renders a single external secret with no overrides:
  1: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        helm.sh/chart: basic-config-0.2.3
      labels:
        app.kubernetes.io/managed-by: Helm
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app-external-secret
      name: test-app-external-secret
      namespace: test-namespace
    spec:
      dataFrom:
        - extract:
            key: test-namespace/test-app/external-secret
      refreshInterval: 5m0s
      secretStoreRef:
        kind: SecretStore
        name: aws-secrets-manager-local
      target:
        creationPolicy: Owner
Check renders multiple external secrets:
  1: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        helm.sh/chart: basic-config-0.2.3
      labels:
        app.kubernetes.io/managed-by: Helm
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app-external-secret1
      name: test-app-external-secret1
      namespace: test-namespace
    spec:
      dataFrom:
        - extract:
            key: test/external/secret1
      refreshInterval: 5m0s
      secretStoreRef:
        kind: SecretStore
        name: aws-secrets-manager-local
      target:
        creationPolicy: Owner
  2: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        helm.sh/chart: basic-config-0.2.3
      labels:
        app.kubernetes.io/managed-by: Helm
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app-external-secret2
      name: test-app-external-secret2
      namespace: test-namespace
    spec:
      dataFrom:
        - extract:
            key: test/external/secret2
      refreshInterval: 5m0s
      secretStoreRef:
        kind: SecretStore
        name: aws-secrets-manager-local
      target:
        creationPolicy: Owner
Check secret name override:
  1: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        helm.sh/chart: basic-config-0.2.3
      labels:
        app.kubernetes.io/managed-by: Helm
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: external-secret1
      namespace: test-namespace
    spec:
      dataFrom:
        - extract:
            key: test/external/secret1
      refreshInterval: 5m0s
      secretStoreRef:
        kind: SecretStore
        name: aws-secrets-manager-local
      target:
        creationPolicy: Owner
Check secret path override:
  1: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        helm.sh/chart: basic-config-0.2.3
      labels:
        app.kubernetes.io/managed-by: Helm
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app-external-secret1
      name: test-app-external-secret1
      namespace: test-namespace
    spec:
      dataFrom:
        - extract:
            key: test/external/secret1
      refreshInterval: 5m0s
      secretStoreRef:
        kind: SecretStore
        name: aws-secrets-manager-local
      target:
        creationPolicy: Owner
